### PR TITLE
Vdb 1067 fk addresses

### PR DESCRIPTION
**Pull request by yaoandrew**
Going forward I'll do one of these per PR. This PR contains:
spot_file_pip
newCDP
vat_suck

### CODE REVIEW

File: transformers/events/new_cdp/transformer_test.go [hunk 1]
```diff
@@ -26,13 +28,25 @@ import (
 )
 
 var _ = Describe("NewCdp Transformer", func() {
-	var transformer = new_cdp.Transformer{}
+	var (
+		transformer = new_cdp.Transformer{}
+		db          = test_config.NewTestDB(test_config.NewTestNode())
```
Comment (rmulhol) [inline transformers/events/new_cdp/transformer_test.go]: I think we probably want to clean the test db in a `BeforeEach` for these tests where we're now initializing a `db`

File: transformers/integration_tests/spot_file_test.go [hunk 2]
```diff
@@ -151,13 +151,17 @@ var _ = Describe("SpotFile EventTransformers", func() {
 
 		It("fetches and transforms a Spot.file pip event", func() {
 			var dbResult spotFilePipModel
-			getSpotErr := db.Get(&dbResult, `SELECT ilk_id, pip from maker.spot_file_pip`)
+			getSpotErr := db.Get(&dbResult, `SELECT ilk_id, what, pip from maker.spot_file_pip`)
```
Comment (rmulhol) [inline transformers/integration_tests/spot_file_test.go]: 👍

File: transformers/test_data/vat_suck.go [hunk 3]
```diff
@@ -54,7 +54,9 @@ var VatSuckEventLog = core.EventLog{
 	Transformed: false,
 }
 
-var VatSuckModel = event.InsertionModel{
+func VatSuckModel() event.InsertionModel { return CopyModel(vatSuckModel) }
```
Comment (rmulhol) [inline transformers/test_data/vat_suck.go]: 🎉

**Review by rmulhol — APPROVED** [review]
LGTM!

Probably worth adding 
```
	BeforeEach(func() {
		test_config.CleanTestDB(db)
	})
```
in the test files where we're now initializing a db, but otherwise no blockers! 👍

File: transformers/events/yank/transformer_test.go [hunk 4]
```diff
@@ -35,6 +35,10 @@ var _ = Describe("Yank Transformer", func() {
 		db          = test_config.NewTestDB(test_config.NewTestNode())
 	)
 
+	BeforeEach(func() {
+		test_config.CleanTestDB(db)
```
Comment (Gslaughl) [inline transformers/events/yank/transformer_test.go]: 🎉

File: transformers/events/new_cdp/transformer.go [hunk 5]
```diff
@@ -71,8 +81,8 @@ func (t Transformer) ToModels(abi string, logs []core.EventLog, _ *postgres.DB)
 			ColumnValues: event.ColumnValues{
 				event.HeaderFK:      newCdpEntity.HeaderID,
 				event.LogFK:         newCdpEntity.LogID,
-				constants.UsrColumn: newCdpEntity.Usr.Hex(),
-				constants.OwnColumn: newCdpEntity.Own.Hex(),
+				constants.UsrColumn: usrAddressID,
+				constants.OwnColumn: ownAddressID,
```
Comment (elizabethengelman) [inline transformers/events/new_cdp/transformer.go]: Wondering if we should change the `usr` and `own` fields in the database to be numerics and foreign keys to the address table.

**Review by elizabethengelman — COMMENTED** [review]
Looking good. The only question I have is if we'd want to change the fields that reference address_ids to be numerics and foreign keys in the database.

**Review by gslaughl — COMMENTED** [review]
Looking good! My only question is whether there should be some kind of corresponding changes in the migrations, where we tell Postgres that these fields are FKs, and probably change the `TEXT` type to an `INTEGER`.

File: transformers/events/new_cdp/transformer.go (hunk 5 above)
Comment (rmulhol) [reply transformers/events/new_cdp/transformer.go]: ➕

**Review by gslaughl — APPROVED** [review]

**Review by elizabethengelman — APPROVED** [review]

**Pull request closed without merging**